demo/addons/fmod/nodes/FmodEventEmitter.gd

```diff
@@ -0,0 +1,80 @@
+tool
+class_name FmodEventEmitter, "res://addons/fmod/nodes/fmod.svg"
+extends Node2D
+
+const UNDEFINED = -1
+const EVENT_PREFIX = "event:/"
+
+export(String) var fmod_event_name = EVENT_PREFIX setget _set_event_name
+export(bool) var attached = true
+export(bool) var autoplay = false
+export(bool) var looped = false
+export(bool) var allow_fadeout = true
+export(Dictionary) var params
+var event_id = UNDEFINED 
+var is_paused = false
+
+func _ready():
+	for key in params:
+		set_param(key, params[key])
+	if autoplay:
+		play()
+		
+func _exit_tree():
+	if event_id != UNDEFINED:
+		if attached:
+			Fmod.detach_instance_from_node(event_id)
+		if allow_fadeout:
+			Fmod.stop_event(event_id, Fmod.FMOD_STUDIO_STOP_ALLOWFADEOUT)
+		else:
+			Fmod.stop_event(event_id, Fmod.FMOD_STUDIO_STOP_IMMEDIATE)
+			
+func set_param(key:String, value:float) -> void:
```

Comment:
I think this method should be split in two because you don't need to reassign value in dictionary from `_ready`